src/main/java/fi/otavanopisto/kuntaapi/server/integrations/casem/CaseMCacheUpdater.java

```diff
@@ -0,0 +1,839 @@
+    
+    for (Long caseMParentId : caseMParentIds) {
+      result.append(String.format("SubNodes(%d)/", caseMParentId));
+    }
+    
+    result.append("SubNodes()");
+    
+    return result.toString();
+  }
+  
+  private Long getSkipToken(String nextLink) {
+    if (StringUtils.isBlank(nextLink)) {
+      return null;
+    }
+    
+    Pattern pattern = Pattern.compile("(.*\\$skiptoken=)([0-9]*)");
+    Matcher matcher = pattern.matcher(nextLink);
+    if (matcher.matches() && (matcher.groupCount() > 1)) {
+      return NumberUtils.createLong(matcher.group(2));
+    }
+    
+    return null;
+  }
+  
+  private Page translateNode(OrganizationId organizationId, Long caseMRootNodeId, Node parentNode, Node node) {
+    Page page = new Page();
+    PageId kuntaApiId = translatePageId(toNodeId(organizationId, node.getNodeId()), true);
+    PageId kuntaApiParentId = null;
+    
+    if (parentNode != null && !parentNode.getNodeId().equals(caseMRootNodeId)) {
+      if (parentNode.getType().equals(0l)) {
+        kuntaApiParentId = parentNode.getParentId() != null ? translatePageId(toNodeId(organizationId, parentNode.getParentId()), false) : null;
+      } else {
+        kuntaApiParentId = translatePageId(toNodeId(organizationId, parentNode.getNodeId()), false);
+      }
+    }
+    
+    List<LocalizedValue> titles = translateNodeNames(organizationId, node.getNames());
+    page.setId(kuntaApiId.getId());
+    page.setParentId(kuntaApiParentId != null ? kuntaApiParentId.getId() : null);
+    page.setTitles(titles);
+    page.setSlug(slugify(titles.isEmpty() ? kuntaApiId.getId() : titles.get(0).getValue()));
+    
+    return page;
+  }
+  
+  private Page translateContent(OrganizationId organizationId, Page parentPage, Content content, String title, String slug) {
+    Page page = new Page();
+    
+    page.setId(translatePageId(toContentId(organizationId, content.getContentId()), true).getId());
+    page.setParentId(parentPage.getId());
+    page.setTitles(toTitles(title));
+    page.setSlug(slug);
+
+    return page;
+  }
+  
+  private PageId translatePageId(PageId pageId, boolean createMissing) {
+    PageId result = discoveredPageIds.get(pageId);
+    if (result != null) {
+      return result;
+    }
+    
+    result = idController.translatePageId(pageId, KuntaApiConsts.IDENTIFIER_NAME);
+    if (result != null) {
+      return result;
+    }
+    
+    if (createMissing) {
+      Identifier identifier = identifierController.createIdentifier(pageId);
+      result = new PageId(KuntaApiConsts.IDENTIFIER_NAME, identifier.getKuntaApiId());
+      discoveredPageIds.put(pageId, result);
+    }
+    
+    return result;
+  }
+  
+  private PageId toNodeId(OrganizationId organizationId, Long nodeId) {
+    return new PageId(CaseMConsts.IDENTIFIER_NAME, String.format("%s|NODE|%d", organizationId.getId(), nodeId));
+  }
+  
+  private PageId toContentId(OrganizationId organizationId, Long contentId) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "toContentId" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)